pkg/volume/local_volume/local_volume.go

```diff
@@ -201,6 +202,34 @@ func (m *localVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
 		return err
 	}
 
+	// If fsGroup is not nil, we need to make sure the requested fsGroup doesn't
+	// conflict with fsGroup already set for previous pod.
+	if fsGroup != nil {
```

Comment:
Hmm if the 2nd pod has no fsGroup, do they run as root? That potentially means whatever files they write to that volume cannot be shared with the 1st pod, which is probably an error?